core/vibes/soul/sections/product-detail/index.tsx

```diff
@@ -8,154 +8,147 @@ import { ProductGallery } from '@/vibes/soul/sections/product-detail/product-gal
 import { ProductDetailForm, ProductDetailFormAction } from './product-detail-form';
 import { Field } from './schema';
 
-interface ProductDetailProduct {
-  id: string;
-  title: string;
-  href: string;
-  images: Streamable<Array<{ src: string; alt: string }>>;
-  price?: Streamable<Price | null>;
-  subtitle?: string;
-  badge?: string;
+interface Props<F extends Field> {
+  productId: string;
+  breadcrumbs?: Streamable<Breadcrumb[]>;
+  title: Streamable<string>;
+  subtitle?: Streamable<string | null>;
   rating?: Streamable<number | null>;
+  price?: Streamable<Price | null>;
+  images: Streamable<Array<{ src: string; alt: string }>>;
   summary?: Streamable<string>;
+  fields: Streamable<F[]>;
   description?: Streamable<string | React.ReactNode | null>;
   accordions?: Streamable<
     Array<{
       title: string;
       content: React.ReactNode;
     }>
   >;
-}
-
-interface Props<F extends Field> {
-  breadcrumbs?: Streamable<Breadcrumb[]>;
-  product: Streamable<ProductDetailProduct | null>;
-  action: ProductDetailFormAction<F>;
-  fields: Streamable<F[]>;
   quantityLabel?: string;
   incrementLabel?: string;
   decrementLabel?: string;
   ctaLabel?: Streamable<string | null>;
   ctaDisabled?: Streamable<boolean | null>;
+  action: ProductDetailFormAction<F>;
   prefetch?: boolean;
   thumbnailLabel?: string;
 }
 
 export function ProductDetail<F extends Field>({
-  product: streamableProduct,
-  action,
-  fields: streamableFields,
+  productId,
   breadcrumbs,
+  title: streamableTitle,
+  subtitle: streamableSubtitle,
+  rating: streamableRating,
+  price: streamablePrice,
+  images: streamableImages,
+  fields: streamableFields,
+  summary: streamableSummary,
+  description: streamableDescription,
+  accordions: streamableAccordions,
```

Comment:
I think we should still take a single `product` prop that is streamable. I think the problem with the current implementation is we have too many fallbacks and skeletons which was implemented that way because of the nested streamables. Instead, we should have a single `product` prop and a single `<Streamable>`